### ISSUE TITLE
Associate file extension with programming language

### ISSUE THREAD

**Issue by peetw** [EWSoftware/VSSpellChecker#98]
Is it possible to associate a file extension with a programming language?

I ask because VSSpellChecker is flagging keywords such as `bool`, `int`, `struct`, `const`, etc. in my C++ files that have a non-standard C++ file extension (they use \*.cu/\*.cuh instead since they are actually CUDA source files). These keywords are not flagged in my "standard" C++ files so I was hoping to be able to associate the \*.cu/\*.cuh file extensions as C++ files in VSSpellChecker.

It should be noted that the keywords are only flagged when performing a spell check via the menu - they are not underlined within the editor window itself. Obviously when the keywords are used in comments they are flagged in both the spell checker and the editor (as expected).

**Comment by EWSoftware** [comment]
If you can give me a list of other extensions besides the two above, I'll update the spell checker classifier configurations for the next release.

In the meantime, you can define a set of custom extensions in your user configuration that map them to the `CStyle` classifier ([Defining Your Own Classifier Configurations](https://github.com/EWSoftware/VSSpellChecker/wiki/0ff35371-69b5-48dd-a062-037abe2469de#defining-your-own-classifier-configurations)).

You'd only need the `Extensions` section to map them to the existing classifier.  For example:

``` XML
<Classifications>
  <Extensions>
    <Extension Value="cu" Classifier="CStyle" />
    <Extension Value="cuh" Classifier="CStyle" />
  </Extensions>
</Classifications>
```

**Comment by peetw** [comment]
Those are the only two file extensions I've seen for CUDA source files. I'm sure someone, somewhere has come up with some special snowflake ones, but those are the "standard" ones.

Thanks for the workaround in the meantime :)

**Comment by EWSoftware** [comment]
I don't have any set plans but I'll probably publish a new release in a few weeks.  I've got a couple of other items to look at too.
